mlrun/run.py

```diff
@@ -1256,716 +1268,161 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
-                    figure = obj.get_figure()
-                elif hasattr(obj, "figure"):
-                    figure = obj.figure
-                elif hasattr(obj, "fig"):
-                    figure = obj.fig
-
-            # Create the artifact:
-            if figure is not None:
-                artifact = PlotArtifact(key=key, body=figure)
-        except ModuleNotFoundError:
-            pass
-
-        # `plotly`:
-        if artifact is None:
-            try:
-                import plotly
-
-                from mlrun.artifacts import PlotlyArtifact
-
-                if isinstance(obj, plotly.graph_objs.Figure):
-                    artifact = PlotlyArtifact(key=key, figure=obj)
-            except ModuleNotFoundError:
-                pass
-
-        # `bokeh`:
-        if artifact is None:
-            try:
-                import bokeh.plotting as bokeh_plt
-
-                from mlrun.artifacts import BokehArtifact
-
-                if isinstance(obj, bokeh_plt.Figure):
-                    artifact = BokehArtifact(key=key, figure=obj)
-            except ModuleNotFoundError:
-                pass
-            except ImportError:
-                logger.warn(
-                    "Bokeh installation is ignored. If needed, "
-                    "make sure you have the required version with `pip install mlrun[bokeh]`"
-                )
-
-        # Log the artifact:
-        if artifact is None:
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                f"The given plot is of type `{type(obj)}`. We currently support logging plots produced by one of "
-                f"the following modules: `matplotlib`, `seaborn`, `plotly` and `bokeh`. You may try to save the "
-                f"plot to file and log it as a file instead."
-            )
-        ctx.log_artifact(**logging_kwargs, item=artifact)
-
-    @staticmethod
-    def log_result(
-        ctx: MLClientCtx,
-        obj: Union[int, float, str, list, tuple, dict, np.ndarray],
-        key: str,
-        logging_kwargs: dict,
-    ):
-        """
-        Log an object as a result. The objects value will be cast to a serializable version of itself. Supporting:
-        int, float, str, list, tuple, dict, numpy.ndarray
+    # Import the module to receive the hinted type:
+    try:
+        # Get the module path and the type class (If we'll wish to support inner classes, the `rsplit` won't work):
+        module_path, type_hint = type_hint.rsplit(".", 1)
+        # TODO: Enabled commonly used aliases (np for numpy, pd for pandas, etc)?
+        # Replace alias if needed (alias assumed to be imported already, hence we look in globals):
+        # For example, if: import A.B.C as abc
+        module_path = module_path.split(".")
+        # And user gave a type hint "abc.Something" then: module_path[0] = "abc"
+        if module_path[0] in globals():
+            # We will replace it with: module_path[0] = "A.B.C"
```

Comment:
not sure I understand why `abc` would get split to `A.B.C`

Reply:
because as mentioned in the comment above, the user did in the handler's code: `import A.B.C as abc`. That means he gave an alias named `abc` to `A.B.C`, hence in `globals` you may find `abc`. Then he gave a type hint called `"some_input_key : abc.Something"` to the `run` method of the mlrun function. So in order to correctly import `Something`, we need to translate the alias, if it appears in `globals` that means it is indeed an alias, otherwise its a full module name.